### DISCUSSION TITLE
@hono/oauth-providers - Github provider returns user email as null

### DISCUSSION THREAD

**Discussion by tinypell3ts** [honojs/hono#2353] [Q&A]
Hi, 

I'm using the [@hono/oauth-providers](https://github.com/honojs/middleware/tree/main/packages/oauth-providers) middleware to manage Github OAuth for my API. Everything is functioning correctly, except the user email is returned as `null`. I have set the permissions for my Github App correctly (as shown below). When testing this app with `next-auth`, the email is returned as expected.

<img width="734" alt="Screenshot 2024-03-14 at 15 16 08" src="https://github.com/honojs/hono/assets/7047410/878d2c89-dc53-4124-8d1f-026ec8635194">

Does anyone have any ideas?

Thanks

**Comment by monoald** [comment]
Hi, @tinypell3ts!

That's weird. I just tested it with a new OAuth app, and I was get the user's email. Could you please send me how you set up the middleware?

**Reply by tinypell3ts** [reply]
Hey @monoald ,

I copied the [example](https://github.com/honojs/middleware/tree/main/packages/oauth-providers#github-app-example) straight from the README.

**Reply by monoald** [reply]
Ooh, I see my error. It gives me the email because I have a "public email". If that is not the case, the email value will be null. I'll fix it. It will just be a matter of time to make the PR and for it to be accepted.

**Reply by tinypell3ts** [reply]
Awesome, thanks @monoald, I'll look out for your PR.

**Comment by tinypell3ts** [comment]
@monoald - I'm struggling to understand how to use the [@hono/oauth-providers](https://github.com/honojs/middleware/tree/main/packages/oauth-providers)(Github) with my NextJS frontend. Are you able to provide some insight into how this works?

**Reply by monoald** [reply]
Of course, how can I help you?

**Reply by tinypell3ts** [reply]
~I'm using a Github App not Oauth App. I'm trying to get an `access_token` and `refresh_token`. I have the `access_token`, but the `refresh_token` is undefined. In the [Authentication Flow](https://github.com/honojs/middleware/tree/main/packages/oauth-providers#authentication-flow-2) section of the docs, it states the `refresh_token` is only available for Oauth Apps. Is this a limitation of the middleware or Github Apps?~ No worries, I've figured it out. The middleware does return a `refresh_token` for GitHub Apps when I opt-in to User-to-server token expiration. Thanks.
